api/src/workers/sale-offer-handler.ts

```diff
@@ -1,30 +1,36 @@
 import { BlockCursorService } from "../services/block-cursor";
 import { FlowService } from "../services/flow";
-import { EventDetails, BaseEventHandler } from "./base-event-handler";
 import { MarketService } from "../services/market";
 
-interface SaleOfferCreated {
-  itemID: number;
-  price: number;
-}
+import { EventDetails, BaseEventHandler } from "./base-event-handler";
+
+import { getConfig } from "../config";
 
 class SaleOfferHandler extends BaseEventHandler {
+  private readonly config;
   constructor(
+    private readonly marketService: MarketService,
     blockCursorService: BlockCursorService,
     flowService: FlowService,
-    private readonly marketService: MarketService,
-    eventName: string
+    eventNames: string[]
   ) {
-    super(blockCursorService, flowService, eventName);
+    super(blockCursorService, flowService, eventNames);
+    this.config = getConfig();
```

Comment:
Yes. This felt awkward when I wrote it. Curious about what side-effects a for the `getConfig` function? 

I think `super()` has to be the first statement in the constructor so this might not work, but coupling the events to this class is more consistent so will try to make that work.

Reply:
Ok, I updated the class. Let me know what you think.